tests/migrations/test_state.py

```diff
@@ -908,6 +909,37 @@ class Meta:
         self.assertEqual([name for name, mgr in food_state.managers], ['food_mgr'])
         self.assertEqual(food_state.managers[0][1].args, ('a', 'b', 1, 2))
 
+    def test_metaclass_conflict(self):
+        """
+        Test making class with complex metadata inheritance
+        """
+        new_apps = Apps(['migrations'])
+
+        class DummyMetaClass(type):
+            dummy_attr = 'dummy'
+
+        class Mixin(six.with_metaclass(DummyMetaClass, object)):
+            mixin_attr = 'mixin'
+
+        class IntermediateMetaClass(DummyMetaClass, ModelBase):
```

Comment:
I don't know if it's public API, but this is Django Models base metaclass, and there is no other way to do it then use it directly.